test/extended/router/gatewayapicontroller.go

```diff
@@ -347,15 +338,15 @@ func checkGatewayStatus(oc *exutil.CLI, gwname, ingressNameSpace string) (*gatew
 	})
 
 	if waitErr != nil {
-		return nil, fmt.Errorf("timed out waiting for gateway %q to become programmed: %w", gateway.Name, waitErr)
+		return nil, fmt.Errorf("Timed out waiting for gateway %q to become programmed: %w", gateway.Name, waitErr)
 	}
 	e2e.Logf("Gateway %q successfully programmed!", gateway.Name)
 	return gateway, nil
 }
 
 // buildGateway initializes the Gateway and returns its address.
 func buildGateway(name, namespace, gcname, fromNs, domain string) *gatewayapiv1.Gateway {
-	hostname := gatewayapiv1.Hostname("*." + "gwapi." + domain)
+	hostname := gatewayapiv1.Hostname("*." + domain)
```

Comment:
Needn't to insert `gwapi.` to the hostname, otherwise it looks like
```
  spec:
    gatewayClassName: openshift-default
    listeners:
    - allowedRoutes:
        namespaces:
          from: All
      hostname: '*.gwapi.gw-default.ci-ln-3if6zbk-72292.origin-ci-int-gce.dev.rhcloud.com'
```
Indeed, we replace `apps.` to `gw-xxx.` in the test to generate the domain for gateway.